terraform/data_provider_ingest_lambda.tf

```diff
@@ -0,0 +1,152 @@
+data "aws_s3_bucket_object" "lambda_zip" {
+  bucket = var.lambda_binary_bucket
+  key = "sdc-dot-cvp-ingest/data_processor.zip"
+}
+
+resource "aws_lambda_function" "IngestLambda" {
+    count = length(var.data_providers)
+    s3_bucket = data.aws_s3_bucket_object.lambda_zip.bucket
+    s3_key = data.aws_s3_bucket_object.lambda_zip.key
+    s3_object_version = data.aws_s3_bucket_object.lambda_zip.version_id
+    function_name = "${var.environment}-dot-sdc-${var.data_providers[count.index]["name"]}-manual-ingest"
+    role = aws_iam_role.IngestLambdaRole[count.index].arn
+    handler = "data_processor.lambda_handler"
+    runtime = "python3.7"
+    timeout = 900 # apparently some of these files get LARGE and can take a while to copy over
+    tags = merge({Name = var.data_providers[count.index]["ingest_bucket"],
+                  Team = var.data_providers[count.index]["team"],
+                  Project = var.data_providers[count.index]["project"]}, local.team_global_tags)
+    environment {
+        variables = {
+            TARGET_DATA_BUCKET = var.data_lake_bucket
+            BUCKET_PATH_MAPPING = jsonencode(map(var.data_providers[count.index]["ingest_bucket"], var.data_providers[count.index]["data_lake_destination"]))
+        }
+    }
+    vpc_config {
+        security_group_ids = [aws_security_group.lambda_http_egress.id]
+        subnet_ids = var.subnet_ids
+    }
+}
+
+resource "aws_cloudwatch_metric_alarm" "IngestLambdaErrors" {
```

Comment:
Interesting side note... the actual email subscriptions on the SNS topic require a manual step so Terraform doesn't report creating them. I set this up so that you can provide an SNS ARN to send emails out.